build/BUILD.bazel

```diff
@@ -38,15 +38,78 @@ install_lualib_deps_cmd = "\n".join([
     for dep in lualib_deps
 ])
 
+wasmx_vm_deps = select({
+    "@kong//:wasmx_v8": [
+        "@v8//:lib",
+    ],
+    "@kong//:wasmx_wasmer": [
+        "@wasmer//:lib",
+    ],
+    "@kong//:wasmx_wasmtime": [
+        "@wasmtime//:lib",
+    ],
+    "//conditions:default": [],
+})
+
+wasmx_deps = select({
+    "@kong//:wasmx_flag": [
+        "@ngx_wasm_module//:lua_libs",
+    ],
+    "//conditions:default": [],
+}) + wasmx_vm_deps
+
+wasm_libs_install = select({
+    "@kong//:wasmx_flag": """
+    for fname in $(locations @ngx_wasm_module//:lua_libs); do
+        base=${fname##*/ngx_wasm_module/lib/}
+        dest="${BUILD_DESTDIR}/openresty/lualib/$base"
+        mkdir -p "$(dirname "$dest")"
+        cp -v "$fname" "$dest"
+    done
+""",
+    "//conditions:default": "\n",
+})
+
+wasmx_vm_cmd = select({
+    "@kong//:wasmx_v8": """
+    if [[ -d ${BUILD_DESTDIR}/openresty/nginx/lib ]]; then
+        copy_with_filter ${BUILD_DESTDIR}/openresty/nginx/lib ${BUILD_DESTDIR}/kong/lib
+        rm -rf ${BUILD_DESTDIR}/openresty/nginx/lib
+    fi
+""",
+    "@kong//:wasmx_wasmer": """
+    if [[ -d ${BUILD_DESTDIR}/openresty/nginx/lib ]]; then
+        copy_with_filter ${BUILD_DESTDIR}/openresty/nginx/lib ${BUILD_DESTDIR}/kong/lib
+        rm -rf ${BUILD_DESTDIR}/openresty/nginx/lib
+    fi
+""",
+    # both v8 and wasmer currently depend on openresty/nginx/lib/libngx_wasm_rs.so,
+    # but in the case of wasmtime it is statically linked and thus not needed in
+    # the final package
+    "@kong//:wasmx_wasmtime": """
+    if [[ -d ${BUILD_DESTDIR}/openresty/nginx/lib ]]; then
+        rm -rf ${BUILD_DESTDIR}/openresty/nginx/lib
+    fi
+""",
+    "//conditions:default": "",
+})
+
+link_modules_dir = select({
+    "@kong//:nginx_dynamic_module_support": """
+    LN ${BUILD_DESTDIR}/openresty/nginx/modules ${BUILD_DESTDIR}/kong/modules
+""",
+    "//conditions:default": "",
+})
+
 kong_directory_genrule(
     name = "kong",
     srcs = [
-        "@openresty//:openresty",
-        "@openresty//:luajit",
         "@luarocks//:luarocks_make",
         "@luarocks//:luarocks_target",
+        "@openresty",
+        "@openresty//:luajit",
```

Comment:
@locao @gszr do you recall if there is a reason the order of these was flipped?

Reply:
It seems to me these lines were moved by `buildifier`.